### DISCUSSION TITLE
Pairing issues

### DISCUSSION THREAD

**Discussion by mrblond18** [Koenkk/zigbee2mqtt#14129] [Devices]
Hi all, after hearing a talk by Koen Kanters himself at a Tweakers meetup I couldn't resist the temptation to jump into zigbee2mqtt... I was using a Philips Hue v1 bridge and a Ikea Tradfri bridge for quite a while and this seems like the perfect way to get rid of those devices.

I bought a "_SONOFF ZBDongle-P Zigbee 3.0 USB Dongle Plus_" dongle and connected it to my Intel NUC running Ubuntu 20.02. After setting zigbee2mqtt up via a docker I got it up and running quite quickly. My Tradfri lightbulbs were immediately found after resetting them and they were immediately loaded into homeassistant as well. Me happy so far!

However, I am trying to add my other devices as well and they just do not seem to be recognized by zigbee2mqtt.

First of all, I have two "simple" Philips Hue lightbulbs, Hue white A60 bulb E27 (https://www.zigbee2mqtt.io/devices/9290011370B.html). I disconnect them using the Philips Hue App, but they are not detected in zigbee2mqtt. When I look for new lights in the Philps Hue app, they are immediately found. When I try to do a touch link scan, I get an error:
```
022-09-21 22:45:51Start Touchlink scan
2022-09-21 22:46:03Request 'zigbee2mqtt/bridge/request/touchlink/scan' failed with error: 'SRSP - AF - interPanCtl after 6000ms'
2022-09-21 22:46:03MQTT publish: topic 'zigbee2mqtt/bridge/response/touchlink/scan', payload '{"data":{},"error":"SRSP - AF - interPanCtl after 6000ms","status":"error","transaction":"oo9d2-2"}'
```

Further, I have a Tradfri wireless dimmer IKEA E2001/E2002 (https://www.zigbee2mqtt.io/devices/E2001_E2002.html) which shows similar behaviour. On that dimmer I can reset it by pressing the link-button four times. Zigbee2mqtt does not see it afterwards, but the Tradfri app does.

What am I doing wrong?

Some details of my setup:
ubuntu 2020.02
zigbee2mqtt running in docker
Zigbee2MQTT version [1.27.2] commit: [7dc48fb]
Coordinator type     zStack3x0
Coordinator revision     20210708
Frontend version     0.6.110

**Comment by Koenkk** [comment]
Nice that you attended the talk!

The coordinator fw you are running has a touchlink bug making the coordinator crash. This is fixed in the latest fw, you can find the latest fw on https://www.zigbee2mqtt.io/guide/adapters/#recommended

**Reply by mrblond18** [reply]
Hi Koen, thanks for the answer. I updated my zigbee stick yesterday with the updated coordinator FW and now I was able to connect to my Philips Hue bulbs after doing touchlink reset, thanks!

Still, I am not able to connect my Ikea E2002, it is very similar to the issue in: https://github.com/Koenkk/zigbee2mqtt/issues/11475
Especially since I am not able to (re)-pair it to my Tradfri Hub either... So it might be that the dimmer itself is the issue, not the Zigbee2MQTT+stick.

**Reply by Koenkk** [reply]
I suggest to take the battery out for a few seconds. It is also recommended to turn off any other Zigbee networks especially if the device has been previously paired to that.

**Reply by mrblond18** [reply]
@Koenkk thanks, I got it working by a lot of trial and error. I must admit I do not understand/remember what did the trick, but it was a lot of pushing the "link button" on the Ikea remote and suddenly it was picked up by zigbee2mqtt.

I have another Ikea remote, the IKEA E1743, which also did not want to pair, but again after a lot of trying I succeeded. Strange thing is that both my IKEA E1524/E1810 remotes immediately coupled after resetting. Anyway, I am quite a few steps further.

There is another thing I am struggling with and I hope you can help. I bound 4 LED1623G12 Ikea bulbs to my Ikea E2002 remote. That worked fine in switching them on and off and controlling brightness. To test my other remote I directly coupled that other remote for testing purposes to one of the bulbs by holding it near the bulb and holding down the pairing button. That way I checked that the remote worked. Now, I had to re-pair the bulb to zigbee2mqtt, that went fine. But binding the bulb back to the Ikea E2002 remote is not working. It is listed as a bind for that remote, but the bulb itself is not reacting. Also, I can not remove the bind or re-create the bind (which in the end would give a double bind I guess). Each time I made sure the remote was "active" by pressing a button before binding.

The error does not tell me much, only a timeout:
```
Debug 2022-09-26 08:47:41Received MQTT message on 'zigbee2mqtt/bridge/request/device/unbind' with data '{"clusters":["genOnOff","genLevelCtrl"],"from":"TuinLampenRemote/1","to":"TuinLampLinksVoor2/1","transaction":"t4wla-1"}'

Debug 2022-09-26 08:47:41unbinding cluster 'genOnOff' from 'TuinLampenRemote' to 'TuinLampLinksVoor2'
Error 2022-09-26 08:48:09Failed to unbind cluster 'genOnOff' from 'TuinLampenRemote' to 'TuinLampLinksVoor2' (Error: Unbind 0xccccccfffe37bb92/1 genOnOff from '0x000b57fffec3068c/1' failed (AREQ - ZDO - unbindRsp after 10000ms))

Debug 2022-09-26 08:48:09unbinding cluster 'genLevelCtrl' from 'TuinLampenRemote' to 'TuinLampLinksVoor2'
Error 2022-09-26 08:48:19Failed to unbind cluster 'genLevelCtrl' from 'TuinLampenRemote' to 'TuinLampLinksVoor2' (Error: Unbind 0xccccccfffe37bb92/1 genLevelCtrl from '0x000b57fffec3068c/1' failed (AREQ - ZDO - unbindRsp after 10000ms))

Info 2022-09-26 08:48:19MQTT publish: topic 'zigbee2mqtt/bridge/response/device/unbind', payload '{"data":{"clusters":[],"failed":["genOnOff","genLevelCtrl"],"from":"TuinLampenRemote/1","to":"TuinLampLinksVoor2/1"},"error":"Failed to unbind","status":"error","transaction":"t4wla-1"}'

Error 2022-09-26 08:48:19Failed to unbind
```

Further, if I look at the light I try to bind to the remote, I see that this bulb is not bound to anything, also not to the coordinator, which the other bulbs are...

Do you have advice?

**Reply by Koenkk** [reply]
You should bind the remote to the bulb, not the other way around. Since the remotes are sleeping most of the times it helps to wake them up (by clicking on them) right before pressing the bind button

**Reply by mrblond18** [reply]
yes, It works now. I thought I did that before, but apparently not (correctly). Thanks (again). Gonna play with window sensors now :-)
